subsys/bluetooth/host/gatt.c

```diff
@@ -786,31 +913,35 @@ static bool update_range(u16_t *start, u16_t *end, u16_t new_start,
 	return true;
 }
 
-static void sc_indicate(struct gatt_sc *sc, u16_t start, u16_t end)
+static void sc_indicate(struct bt_conn *conn, u16_t start, u16_t end)
 {
-	if (!atomic_test_and_set_bit(sc->flags, SC_RANGE_CHANGED)) {
-		sc->start = start;
-		sc->end = end;
+	BT_DBG("start 0x%04x end 0x%04x", start, end);
+
+	if (!atomic_test_and_set_bit(gatt_sc.flags, SC_RANGE_CHANGED)) {
+		gatt_sc.start = start;
+		gatt_sc.end = end;
 		goto submit;
 	}
 
-	if (!update_range(&sc->start, &sc->end, start, end)) {
+	if (!update_range(&gatt_sc.start, &gatt_sc.end, start, end)) {
 		return;
 	}
 
 submit:
-	if (atomic_test_bit(sc->flags, SC_INDICATE_PENDING)) {
+	if (atomic_test_bit(gatt_sc.flags, SC_INDICATE_PENDING)) {
 		BT_DBG("indicate pending, waiting until complete...");
 		return;
 	}
 
 	/* Reschedule since the range has changed */
-	k_delayed_work_submit(&sc->work, SC_TIMEOUT);
+	k_delayed_work_submit(&gatt_sc.work, SC_TIMEOUT);
 }
+#endif /* CONFIG_BT_GATT_SERVICE_CHANGED */
 
+#if defined(CONFIG_BT_GATT_DYNAMIC_DB)
+#if defined(CONFIG_BT_GATT_CACHING)
 static void db_changed(void)
 {
-#if defined(CONFIG_BT_GATT_CACHING)
```

Comment:
I rather have it defined here instead of around each time we call db_changed, not sure why you have decided to change it?

Reply:
I thought it would be best to just ifdef out everything that had to do with a specific option when not selected, makes it easier to understand for me, but that's a matter of opinion. :)

Reverted in latest push.